Engine/Visa.cs

```diff
@@ -4,746 +4,1043 @@
+        private delegate int viUninstallHandlerDelegate(int vi, int eventType, IVisa.viEventHandler handler, int userHandle);
+        private delegate int viInstallHandlerDelegate2(int vi, int eventType, IVisa.viEventHandler handler, int UserHandle);
+        private delegate int viUninstallHandlerDelegate2(int vi, int eventType, IVisa.viEventHandler handler, int userHandle);
+        private unsafe delegate int viReadDelegate(int vi, ArraySegment<byte> buffer, int count, out int retCount);
+        private unsafe delegate int viWriteDelegate(int vi, ArraySegment<byte> buffer, int count, out int retCount);
+        private delegate int viReadSTBDelegate(int vi, ref short status);
+        private delegate int viClearDelegate(int vi);
+        private delegate int viLockDelegate(int vi, int lockType, int timeout, string requestedKey, StringBuilder accessKey);
+        private delegate int viUnlockDelegate(int vi);
+
+        private static viOpenDefaultRMDelegate viOpenDefaultRMRef;
+        private static viFindRsrcDelegate viFindRsrcRef;
+        private static viFindNextDelegate viFindNextRef;
+        private static viParseRsrcDelegate viParseRsrcRef;
+        private static viParseRsrcExDelegate viParseRsrcExRef;
+        private static viOpenDelegate viOpenRef;
+        private static viCloseDelegate viCloseRef;
+        private static viGetAttributeDelegate1 viGetAttribute1Ref;
+        private static viGetAttributeDelegate2 viGetAttribute2Ref;
+        private static viGetAttributeDelegate3 viGetAttribute3Ref;
+        private static viSetAttributeDelegate1 viSetAttribute1Ref;
+        private static viSetAttributeDelegate2 viSetAttribute2Ref;
+        private static viStatusDescDelegate viStatusDescRef;
+        private static viEnableEventDelegate viEnableEventRef;
+        private static viDisableEventDelegate viDisableEventRef;
+        private static viInstallHandlerDelegate viInstallHandlerRef;
+        private static viUninstallHandlerDelegate viUninstallHandlerRef;
+        private static viReadDelegate viReadRef;
+        private static viWriteDelegate viWriteRef;
+        private static viReadSTBDelegate viReadSTBRef;
+        private static viClearDelegate viClearRef;
+        private static viLockDelegate viLockRef;
+        private static viUnlockDelegate viUnlockRef;
+        
+        static TraceSource staticLog = OpenTap.Log.CreateSource("Visa");
+        static Visa()
+        {
+            object instance = null;
+            var visaProviders = TypeData.GetDerivedTypes<IVisaProvider>();
+            var instances = new SortedDictionary<double, (string, IVisa)>();
+            foreach (var typeDataIVisa in visaProviders) 
+            {
+                if (typeDataIVisa != null)
+                {
+                    try 
+                    {
+                        staticLog.Debug("Attempting to load IVisaProvider: {0}", typeDataIVisa.Name);
```

Comment:
Maybe we should remove this log message and add one in the catch clause. 

This works best if we dont try to call Visa on low-priority providers.